### DISCUSSION TITLE
v1.108.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#11005] [Announcements]
# v1.108.0

> [!CAUTION]
> This release includes a security update for Node.js. While we don't think these CVEs affect Immich, we strongly recommend you update. For more details, see https://nodejs.org/en/blog/vulnerability/july-2024-security-releases.

## Highlights

Welcome to release `v1.108.0` of Immich. This release focuses on bug fixes and improvements across the app. We added a fun section of things we learned when working on Immich. You can take a look here at the [Cursed Knowledge](https://immich.app/cursed-knowledge) page. Some of the highlights in this release include:

- Notable fix: Use Redis for Websocket state tracking to allow HDD to spin down when inactive
- Using queue to improve CLI processing 

### 🗄️ Server
* fix(server): healthcheck by @jrasm91 in https://github.com/immich-app/immich/pull/10811
* fix(server): Postgres -> Redis websocket by @tlvince in https://github.com/immich-app/immich/pull/10801
* fix(server): face search migration sometimes failing by @mertalev in https://github.com/immich-app/immich/pull/10827
* refactor(server): notification events by @jrasm91 in https://github.com/immich-app/immich/pull/10754
* refactor(server): rename asset stack to stack by @jrasm91 in https://github.com/immich-app/immich/pull/10828
* refactor(server): bulk interface by @jrasm91 in https://github.com/immich-app/immich/pull/10889
* refactor(server): remove has asset by @jrasm91 in https://github.com/immich-app/immich/pull/10891
* refactor(server): stack owner by @jrasm91 in https://github.com/immich-app/immich/pull/10900
* fix(server): do not try to upgrade to an older version by @jrasm91 in https://github.com/immich-app/immich/pull/10903
* refactor(server): use .toSorted by @jrasm91 in https://github.com/immich-app/immich/pull/10904
* fix(server): reset-admin-password command by @michelheusschen in https://github.com/immich-app/immich/pull/10939
* refactor(server): build resources by @jrasm91 in https://github.com/immich-app/immich/pull/10958
* refactor(server): partner search dto by @jrasm91 in https://github.com/immich-app/immich/pull/10902
* fix(server): wrong video rotation with QSV and RKMPP decoding by @apeman76 in https://github.com/immich-app/immich/pull/10905
* fix(server): avoid server error for invalid email data type by @michelheusschen in https://github.com/immich-app/immich/pull/10978
* fix(server): skip smtp validation when there are no changes by @michelheusschen in https://github.com/immich-app/immich/pull/10991
* chore(server): remove unused code by @jrasm91 in https://github.com/immich-app/immich/pull/10999
### 📱 Mobile
* fix(mobile): exposure time display by @waclaw66 in https://github.com/immich-app/immich/pull/10842
* chore(mobile): separate build flavors by @alextran1502 in https://github.com/immich-app/immich/pull/10872
* fix(mobile): search picker overflow by @alextran1502 in https://github.com/immich-app/immich/pull/10870
### 🖥️ Web
* chore(web): remove unnused property by @jrasm91 in https://github.com/immich-app/immich/pull/10820
* fix(web): delete library by @jrasm91 in https://github.com/immich-app/immich/pull/10822
* chore(web): update translations by @weblate in https://github.com/immich-app/immich/pull/10795
* chore: remove unused snowburst one font by @michelheusschen in https://github.com/immich-app/immich/pull/10863
* fix(web): user profile translation by @waclaw66 in https://github.com/immich-app/immich/pull/10851
* feat(web): use browser language by default by @michelheusschen in https://github.com/immich-app/immich/pull/10849
* fix(web): duplicate thumbnail cover full width by @michelheusschen in https://github.com/immich-app/immich/pull/10880
* chore(web): remove unused s function by @michelheusschen in https://github.com/immich-app/immich/pull/10878
* refactor(web): use derived instead of get(t) by @michelheusschen in https://github.com/immich-app/immich/pull/10884
* fix(web): download archive for public user by @michelheusschen in https://github.com/immich-app/immich/pull/10877
* chore(web): update translations by @weblate in https://github.com/immich-app/immich/pull/10831
* chore: add az language by @jrasm91 in https://github.com/immich-app/immich/pull/10890
* fix(web): external library disk usage unit by @michelheusschen in https://github.com/immich-app/immich/pull/10935
* chore(web): translate alt text by @ben-basten in https://github.com/immich-app/immich/pull/10922
* refactor(web): focus trap by @michelheusschen in https://github.com/immich-app/immich/pull/10915
* refactor(web): show & hide people by @michelheusschen in https://github.com/immich-app/immich/pull/10933
* chore(web): update translations by @weblate in https://github.com/immich-app/immich/pull/10896
* feat(web): improve show & hide people accessibility by @michelheusschen in https://github.com/immich-app/immich/pull/10954
* refactor(web): use callbacks for admin setting events by @michelheusschen in https://github.com/immich-app/immich/pull/10997
* feat(web): render component in notifications by @michelheusschen in https://github.com/immich-app/immich/pull/10990
* chore(web): update translations by @weblate in https://github.com/immich-app/immich/pull/10961
### 🧠 Machine Learning
* fix(ml): armnn not being used by @mertalev in https://github.com/immich-app/immich/pull/10929
### ⚡ CLI
* feat(cli): use a queue for duplicate and upload by @masterT in https://github.com/immich-app/immich/pull/10750
### 📓 Documentation
* docs: cursed knowledge by @jrasm91 in https://github.com/immich-app/immich/pull/10907
### 🔨 Maintenance
* chore(deps): bump docker/build-push-action from 6.2.0 to 6.3.0 by @dependabot in https://github.com/immich-app/immich/pull/10805
* chore(deps): update base-image to v20240702 (major) by @renovate in https://github.com/immich-app/immich/pull/10821
* chore(deps): bump docker/setup-qemu-action from 3.0.0 to 3.1.0 by @dependabot in https://github.com/immich-app/immich/pull/10844
* chore(deps): bump docker/setup-buildx-action from 3.3.0 to 3.4.0 by @dependabot in https://github.com/immich-app/immich/pull/10843
* chore(deps): update dependency prettier-plugin-organize-imports to v4 by @renovate in https://github.com/immich-app/immich/pull/10930
* chore(deps): update typescript-projects by @renovate in https://github.com/immich-app/immich/pull/10962
* chore(deps): update dependency node to v20.15.1 by @renovate in https://github.com/immich-app/immich/pull/10969
* chore(deps): update base-image to v20240709 (major) by @renovate in https://github.com/immich-app/immich/pull/10956
* chore(deps): update base-image to v20240710 (major) by @renovate in https://github.com/immich-app/immich/pull/10995
* chore(deps): update typescript-projects by @renovate in https://github.com/immich-app/immich/pull/10975
* chore(deps): update node by @renovate in https://github.com/immich-app/immich/pull/10996
* chore(deps): update prom/prometheus docker digest to f20d312 by @renovate in https://github.com/immich-app/immich/pull/10998
### Other changes
* feat: update dutch translations by @JordyEGNL in https://github.com/immich-app/immich/pull/10897
* chore: pin specific node version in .nvmrc by @zackpollard in https://github.com/immich-app/immich/pull/10973
* chore(mobile): update photo_manager dep by @alextran1502 in https://github.com/immich-app/immich/pull/10965
* fix(mobile): crash when navigate between large video files by @alextran1502 in https://github.com/immich-app/immich/pull/11002

## New Contributors
* @masterT made their first contribution in https://github.com/immich-app/immich/pull/10750
* @apeman76 made their first contribution in https://github.com/immich-app/immich/pull/10905

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.107.2...

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.108.0'>v1.108.0</a>.</em>

**Comment by waclaw66** [comment]
Seems mobile version wasn't updated. It still shows 1.107.2 after update from Play Store.

**Reply by alextran1502** [reply]
Thanks, I just push new update

**Reply by Quadrubo** [reply]
Hey, sorry to annoy but the update is still missing from F-Droid. Did you maybe only push to Google Play?

**Reply by Voklen** [reply]
@Quadrubo F-Droid is always quite slow to update apps, so just wait a few days and it should be out

**Comment by JordyEGNL** [comment]
> Seems mobile version wasn't updated. It still shows 1.107.2 after update from Play Store.

It takes a while before the new update gets approved by Google.

**Reply by waclaw66** [reply]
It is 1.108.0, but with old version number. Google already approved it.

**Reply by JordyEGNL** [reply]
Whoops did not see that!

**Comment by Str1atum** [comment]
Language seems to auto-revert to system language despite manual setting English in the preferences

**Comment by Emeraldiaa** [comment]
Hi im still new to linux and immich. Can someone tell me how i do the update ?

**Reply by diyoyo** [reply]
Hi @Emeraldiaa , if you're using docker compose, you just have to:
- (Optional but recommended): backup your immich data
- turn off your immich server (`docker compose down`) 
- change the version number in your docker-compose file or environment variable file.
- turn back on the immich server (`docker compose up -d`)

Turning it back on will automatically trigger the pulling of the new image.

**Important Note 1**: it won't remove the old docker image. but that's for another topic ;)
**Important Note 2**: this list of steps is only valid because the immich team did not mention something like "breaking update". In that case backing up is not-so-optional ;) and you'll have to follow extra steps, usually mentioned in the feature log.

**Reply by mueez-aizaz** [reply]
Hey Thanks for the reply. As a noob user myself, I have run into trouble multiple times and still don't understand how exactly to upgrade. 

In the documentation, bringing down the docker compose is not mention. There it simply says to use the following command

_docker compose pull && docker compose up -d_

Often times, when I do this I my previously uploaded data, and immich starts from scratch again.

**Comment by AndyPro720** [comment]
Hey, same issue after this update too. I tried the [fix ](https://github.com/immich-app/immich/issues/10767)by @alextran1502 but it hasn't resolved the issue.

```
immich_postgres          | 2024-07-11 10:34:33.777 UTC [301] ERROR:  internal error: entered unreachable code
immich_postgres          | 2024-07-11 10:34:33.777 UTC [301] STATEMENT:
immich_postgres          |            CREATE INDEX IF NOT EXISTS clip_index ON smart_search
immich_postgres          |            USING hnsw (embedding vector_cosine_ops)
immich_postgres          |            WITH (ef_construction = 300, m = 16)
immich_server            | Migration "AddFaceSearchRelation1718486162779" failed, error: internal error: entered unreachable code
immich_server            | QueryFailedError: internal error: entered unreachable code
immich_server            |     at PostgresQueryRunner.query (/usr/src/app/node_modules/typeorm/driver/postgres/PostgresQueryRunner.js:219:19)
...
```
 Thanks

**Reply by mertalev** [reply]
In this case it affects the `smart_search` table, so run this instead:
```sql
ALTER TABLE smart_search ALTER COLUMN embedding SET DATA TYPE real[];
ALTER TABLE smart_search ALTER COLUMN embedding SET DATA TYPE vector(512);
```

**Reply by AndyPro720** [reply]
Thanks @mertalev, I had a feeling on the same. Any idea when this edge case bug would be resolved?

Also, I do I need to run any job again?

**Reply by mertalev** [reply]
> Any idea when this edge case bug would be resolved?

If my understanding of the issue is right, this error should only be possible with embeddings generated in older versions of Immich (< 1.95.0).

> Also, I do I need to run any job again?

Nope!

**Reply by AndyPro720** [reply]
>  this error should only be possible with embeddings generated in older versions of Immich (< 1.95.0).

But wouldn't this be common for folks using immich since long, with many embeddings generated prior. I faced the issue understandably while updating from v106 to 7/8.

**Comment by kjkent** [comment]
I absolutely love the Cursed Knowledge page :laughing:

**Reply by malteo** [reply]
It's a great idea indeed, funny & informative
